src/features/Sync/SyncSlice.ts

```diff
@@ -18,6 +91,51 @@ const { actions: SyncAction, reducer: SyncReducer } = createSlice({
     name: REDUCER.SYNC
 });
 
+const getSliceState = ({ sync }: RootState): SyncSliceState => {
+    return sync;
+};
+
+const getUrl = (state: RootState): string => {
+    const { url } = getSliceState(state);
+    return url;
+};
+
+const getUsername = (state: RootState): string => {
+    const { username } = getSliceState(state);
+    return username;
+}
+
+const getPassword = (state: RootState): string => {
+    const { password } = getSliceState(state);
+    return password;
+}
+
+const SyncSelector = {
+    getUrl,
+    getUsername,
+    getPassword
+};
+
+function* updateUrl({ payload: { url }}: UpdateUrlAction): SagaIterator {
+    const DependencyLocator = yield getContext(DEPENDENCY.LOCATOR);
```

Comment:
assuming persistance storage is later?

We will need urls for each of the endpoints

Reply:
Yup. Ah, ok. I wasn't sure if the endpoints were standardised!

Reply:
just as a standalone app integrating with other systems they will likely define their own I guess 🤷

Reply:
Resolved by https://github.com/openmsupply/msupply-cold-chain/pull/54.